config/main.py

```diff
@@ -4008,7 +4008,8 @@ def add_route(ctx, command_str):
 
     # If defined intf name, check if it belongs to interface
     if 'ifname' in route:
-        if (not route['ifname'] in config_db.get_keys('VLAN_INTERFACE') and
+        if (not route['ifname'] in config_db.get_keys('VLAN') and
```

Comment:
This will break the assumption made as part of PR https://github.com/Azure/sonic-utilities/pull/1535#issuecomment-825330038

Reply:
I see that this will allow configuring a route to ifname without ip, but this seems to be the only way to solve this problem.